py_neuromodulation/nm_stream.py

```diff
@@ -134,23 +140,38 @@ def _run(
                 data_batch.astype(np.float64)
             )
             if is_stream_lsl:
-                feature_dict["time"] = time_[-1]
                 if self.verbose:
                     if last_time is not None:
                         logger.debug("%.3f seconds of new data processed", time_[-1] - last_time)
+                        feature_dict["time"] = time_[-1] - last_time
+                    else:
+                        feature_dict["time"] = 0
                     last_time = time_[-1]
             else:
-                feature_dict["time"] = np.ceil(time_[-1] * 1000 +1 ).astype(int)
+                feature_dict["time"] = np.ceil(time_[-1] * 1000 +1 )
                 logger.info("Time: %.2f", feature_dict["time"]/1000)
             
 
             self._add_target(feature_dict, data_batch)
+            buff_cnt += 1
 
-            l_features.append(feature_dict)
+            feature_dict = db.cast_values(feature_dict)
+            if not data_acquired:
+                db.create_table(feature_dict)
+                data_acquired = True
+            db.insert_data(feature_dict)
 
-        feature_df = pd.DataFrame(l_features)
+            if buff_cnt >= save_interval:
+                db.commit()
+                buff_cnt = 0
 
-        self.save_after_stream(out_path_root, folder_name, feature_df)
+        db.commit()
+        feature_df = db.fetch_all()
+
+
+        self.save_after_stream(out_path_root, folder_name, feature_df, save_csv=save_csv)
+
+        db.close()
 
         return feature_df
 
```

Comment:
Yes, but I think it's still nice interface to kind of see the structure what features were calculated for which channels

Reply:
The problem is, for long recordings, the moment the recording finishes it will try to load the entire database in memory and if it's too big it will hand or crash. Luckily thanks to the db itself this wouldn't cause a data-loss, but it's kinda awkward. For now I have added a `return_df` parameter True by default.